lib/sqlalchemy/sql/functions.py

```diff
@@ -1770,6 +1809,13 @@ class coalesce(ReturnTypeFromOptionalArgs[_T]):
     inherit_cache = True
 
 
+class avg(GenericFunction[decimal.Decimal]):
```

Comment:
I think that by default we could make this return float (as in `Numeric(asdecimal=False)`), since it's what most drivers will do at the moment.

Returning decimal, while more in line with the db type, is likely to cause significant impact to users.

what to you think @zzzeek ?